src/blog/2024/09/how-to-debug-node-red-flows.md

```diff
@@ -0,0 +1,174 @@
+
+![Image showing the messages section in the sidebar](./images/message-section.png){data-zoomable}
+_Image showing the messages section in the sidebar_
+
+2. **Messages**: This section shows any messages currently queued up in the runtime, giving you visibility into the data being processed at various stages of your flow.
+
+![Image showing the controls in the sidebar](./images/debugger-controls.png){data-zoomable}
+_Image showing the controls in the sidebar_
+
+At the top of the Debugger tab, you will find controls to stop the runtime manually and buttons to resume execution and step through the flow one input or output at a time when it is paused.
+
+### Pausing the Runtime Manually and Navigating Through Each Step
+
+Now, let's diagnose the flow. We’ll manually pause the runtime, then step through each part of the flow using the debugger controls, observing the changes at each step.
+
+![Image shows the execution of flow while debugger enabled and how to proceed to subsequent execution](./images/proceeding-further-execution.gif){data-zoomable}
+_Image show the execution of flow while debugger enabled and how to proceed to subsequent execution_
+
+Follow these steps:
+
+1. Go to the **Debugger** tab in the sidebar.
+2. Click the **Pause** button in the top-right corner to halt the runtime.
+3. Next, click the **Inject** button to start the execution of the flow.
+4. Once paused, you'll notice that the flow executes step by step, depending on the total inputs, outputs, and number of messages they produce and the message length. Each message will be printed in the **Messages** section of the debugger tab. At the top of each message, the name of the node that generated it will be displayed.
+5. To proceed, click the **step forward** button (represented as an array icon next to the pause button). As you move forward, the **Messages** field will update with the message sent by each node, and the execution will also resume at the next step. Additionally, the input/output of the node sending the message will be highlighted in the flow with a light-bordered rectangle.
+6. As we progress through the execution, everything works fine up to the **Switch** node, where the message passes through correctly. However, when you reach the **Join** node, the highlighted box does not move forward, and no message is printed in the debugger tab. This indicates the issue lies between the **Switch** node and the **Join** node.
+
+Now that we have identified the problem area through manual pausing and stepping through the flow execution, we can utilize breakpoints. Adding breakpoints to the suspected areas is always recommended, as it saves time and effort. Manual stepping through the flow is useful for demonstration purposes, and going through the manual process first can enhance your understanding of how the flow operates, making it easier to identify where breakpoints should be placed effectively.
```

Comment:
```suggestion
Manually stepping through the flow is useful for understanding how the flow operates, making it easier to identify where breakpoints should be placed effectively.
```

A little bit wordy an repetitive.  I understand you wished to summarise the above steps and lead into adding debug points but it felt a bit "preachy" (for want of a better word)